src/services/pcn-router/src/Ports.cpp

```diff
@@ -159,10 +219,15 @@ void Ports::setIp(const std::string &value) {
     // set new ip
     if (parent_.get_shadow()) {
       int prefix = get_netmask_length(getNetmask());
-      std::function<void()> doThis = [&]{parent_.netlink_instance_router_.set_iface_ip(getName(), value, prefix);};
+      std::function<void()> doThis = [&]{parent_.netlink_instance_router_.add_iface_ip(getName(), value, prefix);};
       polycube::polycubed::Namespace namespace_ = polycube::polycubed::Namespace::open("pcn-" + parent_.get_name());
       namespace_.execute(doThis);
     }
+
+    // Align Port with the peer interface
+    int prefix = get_netmask_length(getNetmask());
+    std::string info_address = ip_ + "/" + std::to_string(prefix);
```

Comment:
info_address is not the best name for this. something like `cidr` could work better.